data/basics/overflow.go

```diff
@@ -17,52 +17,35 @@
 package basics
 
 import (
-	"math"
 	"math/bits"
+
+	"golang.org/x/exp/constraints"
 )
 
 // OverflowTracker is used to track when an operation causes an overflow
-type OverflowTracker struct {
+type OverflowTracker[T constraints.Unsigned] struct {
 	Overflowed bool
 }
 
-// OAdd16 adds 2 uint16 values with overflow detection
-func OAdd16(a uint16, b uint16) (res uint16, overflowed bool) {
-	res = a + b
-	overflowed = res < a
-	return
-}
-
-// OAdd32 adds 2 uint32 values with overflow detection
-func OAdd32(a uint32, b uint32) (res uint32, overflowed bool) {
-	res = a + b
-	overflowed = res < a
-	return
-}
+// OverflowTrackerU64 is the u64 instantiation for OverflowTracker
+type OverflowTrackerU64 = OverflowTracker[uint64]
```

Comment:
I don't like this very much. Seems like creates "noiser" call sites.  Does any of our code want to used `OverflowTracker` on non uint64 types? If not, I'd probably leave it alone.